### PR TITLE
Add enabled pseudo to prevent hover style on disabled links

### LINKED ISSUE #369: Disabled "link" buttons still have a hover state background colour change
Shouldn't change colour on hover when a button of "link" type is disabled.

cc @blazepollard

### PR DESCRIPTION
**Pull request by jaepass**
Closes #369

### CODE REVIEW

File: src/components/ChecButton.vue [hunk 1]
```diff
@@ -323,14 +323,14 @@ export default {
         @apply bg-#{map.get($config, 'default')} #{map.get($config, 'text')};
 
         &:not(.button--disabled):hover,
-        &:not(:disabled):hover {
+        &:not(:disabled):hover:enabled {
```
Comment (robbieaverill) [inline src/components/ChecButton.vue:326]: This seems kind of strange to me, does the `:not(:disabled)` part not work?

Comment (jaepass) [reply src/components/ChecButton.vue:326]: Oddly no, not sure why it disregards the links as span tags but works for buttons

Comment (ScopeyNZ) [reply src/components/ChecButton.vue:326]: Interesting, it's related to the different tag types I guess. Maybe the `:disabled` pseudo-selector doesn't make sense for anchor tags.

Comment (robbieaverill) [reply src/components/ChecButton.vue:326]: Yeah anchors don't have that selector, good point. Maybe we should use classes instead?

**Review by ScopeyNZ — APPROVED** [review]
Seems to work okay to me. :shipit:

File: src/components/ChecButton.vue (hunk 1 above)
Comment (ScopeyNZ) [reply src/components/ChecButton.vue:326]: 🤷 I was just about to merge. It does look a bit weird having both, but CSS doesn't seem to have any issue and it works.

Comment (robbieaverill) [reply src/components/ChecButton.vue:326]: That's fine

**Pull request merged**